scripts/genRef.py

```diff
@@ -393,8 +393,10 @@ def emitPage(baseDir, specDir, pi, file):
             logWarn('emitPage:', pageName, 'INCLUDE is None, no page generated')
             return
 
-        # Specification text
-        lines = remapIncludes(file[pi.begin:pi.include + 1], baseDir, specDir)
+        # Specification text from beginning to just before the parameter
+        # section. This covers the description, the prototype, the version
+        # note, and any additional version note text.
+        lines = remapIncludes(file[pi.begin:pi.param], baseDir, specDir)
```

Comment:
Under some conditions, pi.param can be None (thus the following logic). The reflib.py logic tries to find the 'parameter section' (which is really the member / parameter / enum description / etc. section following the include), but there may not be one - though there should be. So while this may work with the current spec text, and its failure may be a good signal that something needs to be changed, I wouldn't recommend this change just as it stands. Playing with this in the Vulkan spec repo and constructing various problematic test cases to try things out with.

Reply:
This has been open a while and I'd like to merge it - as observed above it seems to work fine with the current spec text.

@oddhack, were you able to find any problematic test cases?

Reply:
I suppose the main argument against this change is that divergence in these scripts between specifications isn't great, and this may not work elsewhere.

```suggestion
        remap_end = pi.include + 1 if pi.param is None else pi.param
        lines = remapIncludes(file[pi.begin:remap_end], baseDir, specDir)
```

The above would fall back to the existing search range if `pi.param` is `None`.  Perhaps that's a pragmatic middle ground?  Though for the OpenCL specification I think having the generation scripts crash on a missing section would be an acceptable outcome.  I haven't actually tested the above change, but for OpenCL it should always take the `else` path.

Reply:
Sorry, missed responding earlier. With @alycm's change this is a no-op on the Vulkan spec repository. If you are happy with it for the OpenCL refpages, please proceed.

Reply:
I've made this change, updated the comment, and checked that the output is unchanged for the OpenCL reference pages.